src/main/java/org/example/AppLauncher.java

```diff
@@ -1,8 +1,27 @@
 package org.example;
 
+import org.example.sum.*;
+import org.example.sum.NumberGetter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
 public class AppLauncher {
+    private static final Logger logger = LoggerFactory.getLogger(AppLauncher.class);
+
     public static void main(String[] args) {
-        SumCalculator sumCalculator = new SumCalculator();
-        System.out.println("Sum of numbers: " + sumCalculator.sum(7));
+        try {
+            NumberGetter userInput = new UserInput();
+            NumberManager numberManager = new NumberManager(userInput);
```

Comment:
```java
public static void main(String[] args) {
    AppLauncher appLauncher = new AppLauncher();
    appLauncher.launch(new UserInput(), new SumCalculator());
}

public void launch(NumberGetter numberGetter, SumCalculator sumCalculator) {
    try {
        NumberManager numberManager = new NumberManager(numberGetter);
        int number = numberManager.getNumericValue();
        logger.info("Retrieved number: {}", number);
        logger.info("Sum of numbers from 1 to {}: {}", number, sumCalculator.sum(number));
    } catch (IllegalStateException | IllegalArgumentException e) {
        logger.error("Error: {}", e.getMessage(), e);
    }
}
```

This would allow you to run tests against the application runner (`Launch`), by passing in which number fetcher you want to use.

E.g. you could write tests like
```
//launcher test using scanner
launch(numberFromScanner)
```

```java
launch(numberFromFile)
```

allowing you to test the whole e2e (end-to-end) application by passing in whichever numberGetter you want.

Reply:
This is really cool idea 😸